client.go

```diff
@@ -0,0 +1,436 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var emailValid = regexp.MustCompile(`^[a-z0-9._%+\-]+@[a-z0-9.\-]+\.[a-z]{2,4}$`)
+
+// InvalidEmail Error
+type InvalidEmail struct{}
+
+func (e InvalidEmail) Error() string {
+	return "Invalid Email format"
+}
+
+// RoleMissing Error
+type RoleMissing struct{}
+
+func (e RoleMissing) Error() string {
+	return "Missing Role Parameter"
+}
+
+// NameMissing Error
+type NameMissing struct{}
+
+func (e NameMissing) Error() string {
+	return "Missing Name Parameter"
+}
+
+// EmailMissing Error
+type EmailMissing struct{}
+
+func (e EmailMissing) Error() string {
+	return "Missing Email Parameter"
+}
+
+// LoginMissing Error
+type LoginMissing struct{}
+
+func (e LoginMissing) Error() string {
+	return "Missing Login Parameter"
+}
+
+// PasswordMissing Error
+type PasswordMissing struct{}
+
+func (e PasswordMissing) Error() string {
+	return "Missing Password Parameter"
+}
```

Comment:
All these missing things should be validated on server.
Please create one error type for all HTTP 422 responces and just copy error message from server to error object model.

Remove all these *Missing